server/PokerTableObjects/PokerTable.js

```diff
@@ -0,0 +1,590 @@
+        + this.tableSeats[3].isPlayerSeatAbleToAct() + ' ' + this.tableSeats[3].madeDecision + '\n'
+        + this.tableSeats[4].isPlayerSeatAbleToAct() + ' ' + this.tableSeats[4].madeDecision + '\n'
+        + this.tableSeats[5].isPlayerSeatAbleToAct() + ' ' + this.tableSeats[5].madeDecision + '\n'
+        );
+
+        for(let i = 0; i < 6; i++){
+            if(this.tableSeats[i].isPlayerSeatAbleToAct()){
+                if(!this.tableSeats[i].madeDecision){
+                    console.log('returning false');
+                    return false; // a player is able to act but has not made a decision
+                }
+            }
+        }
+        return true; // everyone that can act has made a decision
+    }
+
+    revealMoreCommunityCards(){
+        if(this.communityCardsShown === 0){
+            // show three (flop)
+            this.communityCardsShown = 3;
+            return true;
+        }
+        else if(this.communityCardsShown === 3){
+            // show one (turn)
+            this.communityCardsShown = 4;
+            return true;
+        }
+        else if(this.communityCardsShown === 4){
+            // show one (river)
+            this.communityCardsShown = 5;
+            return true;
+        }
+        else{
+            return false; // time for the show down
+        }
+    }
+
+    showOneMoreCommunityCard(){
+        this.communityCardsShown++;
+    }
+
+    // now we need to finish the showdown
+    beginTheShowDown(){
+        // we set the seat states of the table
+        for(let i = 0; i < 6; i++){
+            this.tableSeats[i].madeDecision = false; // reset decision
+            this.tableSeats[i].pot += this.tableSeats[i].bet; // update each player's pot
+            this.potTotal += this.tableSeats[i].bet; // update the total pot
+            this.tableSeats[i].bet = 0; // reset bet
+            this.tableSeats[i].turn = false;
+        }
+
+        for(let i = 0; i < 6; i++){
+            if(this.tableSeats[i].inPlay){
+                this.cardsShownToAllPlayers[i] = ['card_back', 'card_back'];
+            }
+        }
+        console.log('the turn: ' + this.tableSeats[2].turn);
+        // calculate all player card ranks
+        for(let i = 0; i < 6; i++){
+            if(this.tableSeats[i].inPlay){
+                this.tableSeats[i].handRank = this.pokerHandRankCalculator.calculate(this.communityCards, this.tableSeats[i].cards);
+                console.log(i + ' ' + this.tableSeats[i].handRank);
+            }
+        }
+        this.currentPlayerSeatCardReveal = this.dealerSeatID; // the dealer is the first to reveal their cards
+        this.showdown = true;
+        
+        // then we show player cards using time out functions, based on whether that player has better cards
+        // then calculate chip distributions, update player seats
+        // wait five seconds again and begin a new game
```

Comment:
What is this comment describing?

Reply:
my bad that was for me when i was implementing it